ste/jobStatusManager.go

```diff
@@ -61,15 +65,36 @@ func (jm *jobMgr) statusMgrClosed() bool {
 
 /* These functions should not fail */
 func (jm *jobMgr) SendJobPartCreatedMsg(msg JobPartCreatedMsg) {
-	jm.jstm.partCreated <- msg
-	if msg.IsFinalPart {
-		// Inform statusManager that this is all parts we've
-		close(jm.jstm.partCreated)
+	if jm.jstm.partCreated != nil { // Sends not allowed if channel is closed
+		jm.jstm.flagMutex.RLock()
+		jm.jstm.partCreated <- msg
+		jm.jstm.flagMutex.RUnlock()
+
+		if msg.IsFinalPart {
+			jm.jstm.flagMutex.Lock()
+			// Inform statusManager that this is all parts we've
+			close(jm.jstm.partCreated)
+			jm.jstm.partCreated = nil
+			jm.jstm.flagMutex.Unlock()
+		}
 	}
 }
 
 func (jm *jobMgr) SendXferDoneMsg(msg xferDoneMsg) {
-	jm.jstm.xferDone <- msg
+	jm.jstm.flagMutex.RLock()
+	defer jm.jstm.flagMutex.RUnlock()
+	if jm.jstm.isXferDoneClosed {
+		fmt.Println("Cannot send message on closed channel")
```

Comment:
This message will pop up on user console. I guess it will not make any sense to user/customer. Either we can just dump it to logs or put a meaningful message for user. If channel is closed for a valid reason then we can skip showing something to user and just add a log message.